### PR TITLE
feat: Add Gemini Code Assist Customization

### PR DESCRIPTION
**Pull request by cboneti**
## feat: Add Gemini Code Assist Customization

This PR introduces a customization for the Gemini Code Assist on GitHub to improve the quality and consistency of automated code reviews for the Cluster Toolkit.

By adding a `.gemini/` directory with a `styleguide.md` and a default `config.yaml`, we provide Gemini with specific instructions and context about the Cluster Toolkit's best practices, structure, and documentation requirements.

The `styleguide.md` emphasizes:

*   Maintainability, Testing, and Documentation.
*   Blueprint Authoring best practices (e.g., `use` block, module sourcing).
*   Terraform Module Development guidelines.
*   Go Language standards.
*   Critical checks for updating index READMEs (`modules/README.md`, `examples/README.md`) when new modules/examples are added.
*   The importance of clear PR descriptions and test coverage.

This customization aims to make Gemini Code Assist reviews more relevant and helpful, catching potential issues early and reinforcing project conventions.

### CODE REVIEW

File: .gemini/styleguide.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+* **Maintainability:** Code should be easy to understand, modify, and extend.
+* **Testing:** Changes must be well-tested. Encourage comprehensive unit and integration tests.
+* **Documentation:** Ensure documentation is updated, including in-code comments, module READMEs, and index files.
+
+Pay close attention to the following specifics:
+
+1. **Blueprint Authoring (YAML):**
+    * Ensure the `use` block is preferred for module dependencies within blueprints. Explicit variable linking (e.g., `setting = $(module.output)`) should only be used when necessary to resolve ambiguity.
+    * Verify that module sources are correct and the referenced modules exist.
+    * Check for logical grouping of modules within `deployment_groups`.
```
Comment (bytetwin) [inline .gemini/styleguide.md]: I think we should define the logical groupings based on module types eg) network, storage, cluster etc

File: .gemini/styleguide.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    * In-code comments should be clear and explain the *why* not just the *what*.
+    * Module `README.md` files should be clear and provide sufficient information on usage, inputs, and outputs.
+
+5. **Testing:**
+    * New features or bug fixes should ideally be accompanied by tests.
+    * Tests should be clear and cover both happy paths and edge cases.
+    * Encourage the use of the existing testing frameworks and patterns within the project.
+
+6. **PR Description:**
+    * The PR description should clearly explain the purpose of the change and the problem it solves.
```
Comment (bytetwin) [inline .gemini/styleguide.md]: This is useful as most of the PRs have no description

File: .gemini/styleguide.md (hunk 1 above)
Comment (cboneti) [reply .gemini/styleguide.md]: PTAL.

Comment (bytetwin) [reply .gemini/styleguide.md]: Logical groups for `deployment_groups` is not defined. You have only added the folder definitions for modules here - https://github.com/GoogleCloudPlatform/cluster-toolkit/pull/4820/commits/778c7a0dcb54f27a3462fff84fad47d80d9112e9#diff-2d64e32e0ed73fabd653270b96703ac45c54f710133a9ed5882c6d0c6efe6f32L24

Comment (cboneti) [reply .gemini/styleguide.md]: ahh, I see. I am not sure how we should guide it at this point as I am not sure how we can guide an agent to find logical deployment groups...

**Review by bytetwin — APPROVED** [review]
